http/client.go

```diff
@@ -14,30 +14,45 @@ import (
 	"time"
 )
 
-// New creates a new http.Client with a custom certificate, which can be loaded from the passed CA Bundle file and/or
+type TransportOption func(*http.Transport) error
+
+// New creates a new http.Client with the Passed Transport Options
+// that will have a custom certificate if it is loaded from the passed CA Bundle file and/or
 // directory. If both CABundleFile and CABundleDir are empty arguments, it creates an unsecure HTTP client.
-func New(CABundleFile, CABundleDir string, httpTimeout time.Duration) (*http.Client, error) {
-	return _new(CABundleFile, CABundleDir, httpTimeout, "")
-}
+func New(httpTimeout time.Duration, opts ...TransportOption) (*http.Client, error) {
+	t := &http.Transport{
+		TLSClientConfig: &tls.Config{},
+	}
+	for _, opt := range opts {
+		if err := opt(t); err != nil {
+			return nil, err
+		}
+	}
 
-// NewAcceptInvalidHostname new http.Client with ability to accept HTTPS certificates that don't
-// match the hostname of the server they are connecting to.
-func NewAcceptInvalidHostname(CABundleFile, CABundleDir string, httpTimeout time.Duration, hostname string) (*http.Client, error) {
-	return _new(CABundleFile, CABundleDir, httpTimeout, hostname)
+	return &http.Client{
+		Timeout:   httpTimeout,
+		Transport: t,
+	}, nil
 }
 
-func _new(CABundleFile, CABundleDir string, httpTimeout time.Duration, acceptInvalidHostname string) (*http.Client, error) {
-	// go default http transport settings
-	t := &http.Transport{}
-
-	if CABundleFile != "" || CABundleDir != "" {
-		certs, err := getCertPool(CABundleFile, CABundleDir)
-		if err != nil {
-			return nil, err
+func WithCABundle(CABundleFile, CABundleDir string) TransportOption {
```

Comment:
this should be 2 different options, right?
I mean you setup the folder or the file, so either load the provided file or read all the certs from the folder

> there's the option to use the folder arg as a prefix path to the file, but I don't think this should be used at all

Reply:
I agree it's strange and thought the same, but if you check the actual logic, nothing it's avoiding you from setting both  at the same time. there is no return statement on the first `if certFile != "" { }` to stop executing the second block where the Dir is red, should I change it or it's a feature?

Anyway, even if we continues with the same logic where you can add both a file and a dir at the same time, I can separate WithCABundleDir and WithCABundleFile and pass them the x509.NewCertPool().  What do you think?

Reply:
+1 to the idea of splitting them. My understanding is that the code just checks for both options and appends them, so we better separate it to avoid uglyness and passing empty strings.